rotkehlchen/chain/evm/decoding/balancer/balancer_cache.py

```diff
@@ -0,0 +1,203 @@
+    - RemoteError
+    """
+    version: Final = 1 if cache_type == CacheType.BALANCER_V1_POOLS else 2
+    protocol = CPT_BALANCER_V1 if version == 1 else CPT_BALANCER_V2
+    latest_pools_count = query_balancer_pools_count(
+        chain=inquirer.chain_id,
+        version=version,
+    )
+    gauge_key_parts = (str(inquirer.chain_id.value), str(version))
+    pool_key_parts = (str(inquirer.chain_id.value),)
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        existing_pools = {
+            string_to_evm_address(address)
+            for address in globaldb_get_general_cache_values(
+                cursor=cursor,
+                key_parts=(cache_type, *pool_key_parts),
+            )
+        }
+        existing_gauges = {
+            string_to_evm_address(address)
+            for address in globaldb_get_general_cache_values(
+                cursor=cursor,
+                key_parts=(CacheType.BALANCER_GAUGES, *gauge_key_parts),
+            )
+        }
+
+    if latest_pools_count == len(existing_pools):
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_update_cache_last_ts(
+                write_cursor=write_cursor,
+                key_parts=pool_key_parts,
+                cache_type=cache_type,
+            )
+
+        return existing_pools, existing_gauges
+
+    pools, gauges = set(), set()
+    for pool in query_balancer_pools(chain=inquirer.chain_id, version=version):
+        try:
+            pool_token = get_or_create_evm_token(
+                userdb=inquirer.database,
+                chain_id=inquirer.chain_id,
+                protocol=protocol,
+                name=pool['name'].strip(),
+                symbol=pool['symbol'].strip(),
+                decimals=deserialize_int(pool['decimals']),
+                evm_address=deserialize_evm_address(pool['address']),
+                encounter=TokenEncounterInfo(
+                    description=f'Querying {inquirer.chain_name} {protocol} balances',
+                    should_notify=False,
+                ),
+                underlying_tokens=[
+                    UnderlyingToken(
+                        address=get_or_create_evm_token(
+                            userdb=inquirer.database,
+                            chain_id=inquirer.chain_id,
+                            protocol=protocol,
+                            name=token['name'].strip(),
+                            symbol=token['symbol'].strip(),
+                            evm_address=deserialize_evm_address(token['address']),
+                            encounter=TokenEncounterInfo(
+                                description=f'Querying {inquirer.chain_name} {protocol} balances',
+                                should_notify=False,
+                            ),
+                        ).evm_address,
+                        token_kind=EvmTokenKind.ERC20,
+                        weight=FVal(token['weight']) if token.get('weight') else ONE,
```

Comment:
here instead of ONE it should be `len(token in pool['poolTokens'] if pool['address'] != token['address'])`. I beleive we have an assumption that the sum is always ONE. Does it hold here?